owncloud/core
# Hide add to your OC at the public page when sending shares is not all…

…owed by admin


## Description
When ` Allow users on this server to send shares to other servers` is unchecked at the Settings->Admin-Sharing there should be no 'Add to your ownCloud' button
Bonus: switch to EventDispatcher as https://github.com/owncloud/core/pull/31944 allows it

## Related Issue
- Fixes https://github.com/owncloud/enterprise/issues/3291

## How Has This Been Tested?
1. Create a public link
2. Tick Settings->Admin-Sharing ` Allow users on this server to send shares to other servers`
3.  open the public link in another browser
Expected:  'Add to your ownCloud' at the public page

1. Create a public link
2. Untick Settings->Admin-Sharing ` Allow users on this server to send shares to other servers`
3.  open the public link in another browser
Expected:  There is NO 'Add to your ownCloud' at the public page

## Types of changes

## Checklist:
